docs/path-resolution.rst

```diff
@@ -313,6 +315,45 @@ interpreted as absolute paths on disk.
 If the base path itself is relative, it is also interpreted as relative to the current working
 directory of the compiler.
 
+.. _base-path-normalization-and-stripping:
+
+Base Path Normalization and Stripping
+-------------------------------------
+
+When source file paths are specified on the command line, base path affects the source unit names
+assigned to them in compiler's VFS.
+To compute the names, both base path and source file paths must first be converted to a canonical form.
+This ensures that the result is predictable and as platform-independent as possible:
+
+- If a path is relative, it is made absolute by prepending current working directory to it.
+
+  - If the path to the working directory contains symbolic links, they are resolved into actual
+    directories.
+
+- Internal ``.`` and ``..`` segments are collapsed.
+- Platform-specific path separators are replaced with forward slashes.
+- Sequences of multiple consecutive path separators are squashed into a single separator (unless
+  they are the leading slashes of an `UNC path <https://en.wikipedia.org/wiki/Path_(computing)#UNC>`_).
+- If the path includes a root name (e.g. a drive letter on Windows) and the root is the same as the
+  root of the current working directory, the root is replaced with ``/``.
+- Symbolic links in the path itself are **not** resolved.
```

Comment:
In the base path or the file path?

Reply:
In both. Both undergo the same normalization. They must if we want prefix stripping to work reliably.

Reply:
Then please add this to the doc (if not already)

Reply:
But the text above the list already says that all of these are applied to both:

> To compute the names, both base path and source file paths must first be converted to a canonical form.

I can add it but I'd prefer not to because doing it only for this one bullet point would make it look like it's the only one that applies to the base path (while they all do).

Reply:
ah right, then it's fine!